internal/provider/namespace_tags_resource.go

```diff
@@ -0,0 +1,334 @@
+	return &namespaceTagsResource{}
+}
+
+func (r *namespaceTagsResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*client.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *client.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	r.client = client
+}
+
+func (r *namespaceTagsResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_namespace_tags"
+}
+
+func (r *namespaceTagsResource) Schema(ctx context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Manages the complete set of tags for a Temporal Cloud namespace.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The ID of this namespace tags resource.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"namespace_id": schema.StringAttribute{
+				Description: "The ID of the namespace to manage tags for.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"tags": schema.MapAttribute{
+				Description: "A map of tag keys to tag values.",
+				ElementType: types.StringType,
+				Required:    true,
+				Validators: []validator.Map{
+					mapvalidator.SizeAtLeast(1),
+				},
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"timeouts": timeouts.Block(ctx, timeouts.Opts{
+				Create: true,
+				Delete: true,
+			}),
+		},
+	}
+}
+
+func (r *namespaceTagsResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan namespaceTagsModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createTimeout, diags := plan.Timeouts.Create(ctx, defaultCreateTimeout*2)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, createTimeout)
+	defer cancel()
+
+	existingTags, err := getNamespaceTags(ctx, r.client, plan.NamespaceID.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to get current namespace tags", err.Error())
+		return
+	}
+
+	plannedTags, d := getTagsFromMap(ctx, plan.Tags)
+	resp.Diagnostics.Append(d...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	err = r.setNamespaceTags(ctx, plan.NamespaceID.ValueString(), existingTags, plannedTags)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to set namespace tags", err.Error())
+		return
+	}
+
+	resp.Diagnostics.Append(updateTagsModelFromNamespace(ctx, &plan, plan.NamespaceID.ValueString(), plannedTags)...)
```

Comment:
I think you want to check the error here and return before calling state.Set below.